packages/muze-legend/src/legend/gradient-legend.js

```diff
@@ -69,8 +69,10 @@ export default class GradientLegend extends SimpleLegend {
             if (domain[domain.length - 1] > steps[steps.length - 1]) {
                 domainForLegend.push(domain[1]);
             }
+            domainForLegend = domainForLegend.sort((a, b) => a - b);
+            domainForLegend = getInterpolatedArrayData(domain, domainForLegend, context);
         } else {
-            domainForLegend = getInterpolatedData(domain, steps - 1);
+            domainForLegend = getInterpolatedData(domain, steps - 1, context);
```

Comment:
The last line for if and else look almost the same. See if it is possible to remove the else statement.